File: Practice/07/Python/7.py
ch = int(input('1 - длины сторон,2 - координаты вершин = '))
if ch == 1:
    a = float(input('a = '))
    b = float(input('b = '))
    c = float(input('c = '))
    p = (a + b + c) / 2
    s = (p * (p - a) * (p - b) * (p - c)) ** 0.5
    print('S = ', s)
elif ch == 2:
    x1, y1 = map(float, input('A = ').split())
    x2, y2 = map(float, input('B = ').split())
    x3, y3 = map(float, input('C = ').split())
    s = (1 / 2) * (abs((x2 - x1) * (y3 - y1) - (x3 - x1) * (y2 - y1)))
    print('S = ', s)
else:
    print('Неправильно набран номер')

File: Practice/07/C++/7.cpp

#include <iostream>

int main() {
using namespace std;
cout << "1 - po dline, 2 - po koordinatam: ";
int ch;
cin >> ch;
if (ch == 1) {
    int a, b, c;
    long p, s;
    cin >> a >> b >> c;
    p = (a + b + c) / 2;
    s = sqrt(p * ((p - a) + (p - b) + (p - c)));
    cout << s;
    }
else if (ch == 2) {
    int x1, x2, x3, y1, y2, y3;
    long s;
    cin >> x1 >> x2 >> x3 >> y1 >> y2 >> y3;
    s = (1 / 2) * (abs((x2 - x1) * (y3 - y1) - (x3 - x1) * (y2 - y1)));
    cout << s;

}
}
